File: server/models/Result.js
const mongoose = require("mongoose");

const resultSchema = new mongoose.Schema({
  interview: { type: mongoose.Schema.Types.ObjectId, ref: "Interview" },
  scores: {
    resume: Number,
    aptitude: Number,
    technical: Number,
    interview: Number
  },
  feedback: String
}, { timestamps: true });

module.exports = mongoose.model("Result", resultSchema);

File: server/models/result.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from bson import ObjectId

class Scores(BaseModel):
    resume: Optional[float] = None
    aptitude: Optional[float] = None
    technical: Optional[float] = None
    interview: Optional[float] = None

class ResultBase(BaseModel):
    interview: str
    scores: Scores
    feedback: str

class ResultCreate(ResultBase):
    pass

class Result(ResultBase):
    id: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True
        json_encoders = {ObjectId: str}